demo.sh

```diff
@@ -594,16 +530,13 @@ transfer_create() {
     if [ "$reuse_invoice" != 1 ]; then
         _subtit "(recipient) preparing invoice"
         if [ "$witness" = 1 ]; then
-            address_mode="-a"
+            address_mode="--wout"
         else
             [ "$NO_GEN_UTXO" != 1 ] && _gen_utxo "$RCPT_WLT"
             address_mode=""
         fi
-        # not quoting $address_mode so it doesn't get passed as "" if empty
-        # shellcheck disable=2086
-        _trace "${RGB[@]}" -d "$rcpt_data" invoice $address_mode \
-            -w "$RCPT_WLT" -i $iface --amount "$send_amt" "$contract_id" \
-            >$TRACE_OUT
+        _trace "${RGB[@]}" -d "$rcpt_data" invoice "$address_mode" \
+            -w "$RCPT_WLT" "$contract_id" "$send_amt" >$TRACE_OUT
```

Comment:
invoicing now simpler: no need to provide interface name.

Also, no more "amounts"; invoice includes now arbitrary state, encoded as a strict value. Good news: for numbers, it is just like writing the number. And it is now a positional argument

Both `contract_id` and `state` as positional arguments are now optional, allowing creating invoices in form of "pay anything" and "pay any amount under this contract"

Reply:
This is where the script execution always fails, with 

```
error: invalid value '' for '[CONTRACT]': invalid contract reference ''

For more information, try '--help'.

ERROR: command 'rgb-wallet/bin/rgb -n regtest -d data1 invoice  -w wallet_1 contract:Ko_9~W2J-YXwpN8e-ZTY9X2l-PtqSsUN-g2Wu0ij-onReuzw 100' returned a non-zero exit code (transfer 0)
```

Spent a lot of time trying to figure why....

Reply:
Ok, solved this: the reason was that `"$address_type"` didn't worked producing an excessive argument in a form of `''` when a UTXO payment is used. Spent day figuring that out... Pity